crates/ty_project/src/lib.rs

```diff
@@ -293,7 +306,10 @@ impl Project {
 
     /// Opens a file in the project.
     ///
-    /// This changes the behavior of `check` to only check the open files rather than all files in the project.
+    /// This changes the behavior of `check` to only check the open files rather than all files in
+    /// the project if the project's check mode is [`OpenFiles`].
```

Comment:
I think we should remove this behavior and instead always check exactly what `CheckMode` configured. This also allows us to remove the `Option` from `open_files`

Reply:
> This also allows us to remove the `Option` from `open_files`

Does this mean that the `take_open_files` would not be needed and we can mutate the open files directly? The open file set is behind an `Arc`, so might need some workaround to that.

Reply:
I think we still need the `take_open_files` internally to avoid cloning the entire hash set every time a file is opened or closed. This could be a reason for keep using an `Option<Arc<...>` internally but we should change/remove the implicit behavior that the project only checks the open files if `open_file_set` is non-none.

Reply:
I've kept the `Option<Arc<...>>` and updated the logic to use the check mode explicitly and removed the implicit behavior of `None`